packages/flutter/lib/src/material/grid_tile_bar.dart

```diff
@@ -118,9 +121,8 @@ class GridTileBar extends StatelessWidget {
     return new Container(
       padding: padding,
       decoration: decoration,
-      child: new IconTheme.merge(
-        context: context,
-        data: new IconThemeData(color: Colors.white),
+      child: new Theme(
+        data: darkTheme,
```

Comment:
Thanks for spotting -- you're entirely correct; it's not. TIL that the `IconThemeData` associated with a `Theme` doesn't override any `IconTheme` above it. 

While I put together a separate change to refactor `Theme` & `IconTheme`, I've patched this to add back an IconTheme, but eliminate the typography hardcoding.